### PR TITLE
Sort __hf_index_id column

### LINKED ISSUE #1920: Values of `__hf_index_id` in DuckDB indexes are not in the right order
For example the first item in de C4 dataset is supposed to be

```
Beginners BBQ Class Taking Place in Missoula!...
```

This is also the first example when querying the DuckDB index, but it has `__hf_index_id == 1000`

```python
con.sql("select * from data limit 5;")
```

```
┌───────────────┬──────────────────────┬──────────────────────┬────────────────────────────────────────────────────────┐
│ __hf_index_id │         text         │      timestamp       │                          url                           │
│     int64     │       varchar        │       varchar        │                        varchar                         │
├───────────────┼──────────────────────┼──────────────────────┼────────────────────────────────────────────────────────┤
│          1000 │ Beginners BBQ Clas…  │ 2019-04-25T12:57:54Z │ https://klyq.com/beginners-bbq-class-taking-place-in…  │
│          1001 │ Discussion in 'Mac…  │ 2019-04-21T10:07:13Z │ https://forums.macrumors.com/threads/restore-from-la…  │
│          1002 │ Foil plaid lycra a…  │ 2019-04-25T10:40:23Z │ https://awishcometrue.com/Catalogs/Clearance/Tweens/…  │
│          1003 │ How many backlinks…  │ 2019-04-21T12:46:19Z │ https://www.blackhatworld.com/seo/how-many-backlinks…  │
│          1004 │ The Denver Board o…  │ 2019-04-20T14:33:21Z │ http://bond.dpsk12.org/category/news/                  │
└───────────────┴──────────────────────┴──────────────────────┴────────────────────────────────────────────────────────┘
```

```python
con.sql("select * from data where __hf_index_id < 5;")
```

```
┌───────────────┬──────────────────────┬──────────────────────┬────────────────────────────────────────────────────────┐
│ __hf_index_id │         text         │      timestamp       │                          url                           │
│     int64     │       varchar        │       varchar        │                        varchar                         │
├───────────────┼──────────────────────┼──────────────────────┼────────────────────────────────────────────────────────┤
│             0 │ Eimhir Ltd, its af…  │ 2019-04-24T10:02:25Z │ https://www.game-pal.com/signup                        │
│             1 │ Roger & Gallet Thé…  │ 2019-04-21T06:22:38Z │ https://www.farmaline.uk/health/order/roger-gallet-t…  │
│             2 │ Folio: 166v - Feas…  │ 2019-04-19T17:07:24Z │ http://cantus.uwaterloo.ca/chant/481394                │
│             3 │ DIY Co Sleeper For…  │ 2019-04-21T16:15:08Z │ https://www.musely.com/tips/DIY-Co-Sleeper-For-Baby-…  │
│             4 │ Rep. Ilhan Omar (D…  │ 2019-04-26T07:55:58Z │ http://theminnesotasun.com/2019/01/13/ilhan-omar-dec…  │
└───────────────┴──────────────────────┴──────────────────────┴────────────────────────────────────────────────────────┘
```

This would show the wrong row idx when clicking on the example after search (which times out atm though)

Not a big deal for now imo

### PR DESCRIPTION
**Pull request by albertvillanova**
Sort `__hf_index_id` column.

After investigation, ~~I think~~ the root cause of the unsorted column is that `duckdb` uses some kind of parallelism when generating a table from a Parquet file (~~maybe~~ across row groups), calling the 'serial' sequence concurrently. To make the column sorted, I propose to add it after the table is generated.

Note that this column is used by `_transform_row` to generate the `row_idx`, and the latter is used by `get_cell_value` to create the audio/image files. Maybe related to:
- #2158

Fix #1920.

### CODE REVIEW

**Review by AndreaFrancis — APPROVED** [review]

**Comment by albertvillanova** [comment]
@severo I have added a test and checked it was not passing before the fix.

I can confirm that the cause is concurrent reading across row groups of the parquet file.

**Comment by severo** [comment]
awesome

File: services/worker/tests/job_runners/split/test_duckdb_index.py [hunk 1]
```diff
@@ -517,8 +519,20 @@ def test_get_indexable_columns(features: Features, expected: list[str]) -> None:
...
+    pa_table = pa.Table.from_pydict({"text": [f"text-{i}" for i in range(100)]})
+    parquet_path = str(tmp_path / "0000.parquet")
+    pq.write_table(pa_table, parquet_path, row_group_size=2)
+    db_path = str(tmp_path / "index.duckdb")
+    column_names = ",".join(f'"{column}"' for column in pa_table.column_names)
+    with duckdb.connect(db_path) as con:
+        con.sql(CREATE_TABLE_COMMANDS.format(columns=column_names, source=parquet_path))
+    with duckdb.connect(db_path) as con:
+        df = con.sql("SELECT * FROM data").to_df()
+    assert df["__hf_index_id"].is_monotonic_increasing
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply services/worker/tests/job_runners/split/test_duckdb_index.py:538]: I finally added. I'm merging after all tests are green.

[the remark this answers is not in the document]
Comment (AndreaFrancis) [reply services/worker/tests/job_runners/split/test_duckdb_index.py:538]: > But if you think so, I could add an assertion to ensure unicity.
It makes sense to me (not necessary).

**Comment by severo** [comment]
shouldn't we have incremented the job runner version? I know that if we do it, it's a lot of jobs to create (all the datasets) but I understand that potentially all the duckdb files are wrong, no?

**Comment by lhoestq** [comment]
Maybe not worth it imo, except manually refreshing datasets with known issues (like showing the wrong audio file).

Otherwise the row id is only used to select rows in the UI and share the URL no ?

**Comment by lhoestq** [comment]
Also note that we will likely refresh the index once duckdb has stabilized their format next year

**Comment by severo** [comment]
yes, I imagine we will have other occasions to refresh all the duckdb files 😅

**Pull request merged**